tfhe/src/integer/server_key/radix_parallel/comparison.rs

```diff
@@ -77,33 +74,36 @@ impl ServerKey {
             std::mem::swap(&mut block_comparisons_2, &mut block_comparisons);
         }
 
-        block_comparisons.resize_with(lhs.blocks().len(), || self.key.create_trivial(0));
-
-        T::from_blocks(block_comparisons)
+        BooleanBlock::new(
+            block_comparisons
+                .into_iter()
+                .next()
+                .unwrap_or_else(|| self.key.create_trivial(0)),
```

Comment:
no way to use block_comparisons.blocks().get(0).unwrap_or_else() ?

I think the iterating part is something we sometimes use but it looks really odd

Reply:
remove ?

Reply:
I think you used it, but maybe it's more costly than it needs to be

Reply:
or drain, but yes I see your point